controllers/grid/queries/form/QueryForm.inc.php

```diff
@@ -323,6 +323,65 @@ function readInputData() {
 		));
 	}
 
+	/**
+	 * @copydoc Form::validate()
+	 */
+	function validate($callHooks = true) {
+		// Display error if anonymity is impacted, i.e. if:
+		// 1) several blind reviewers are selected, or
+		// 2) a blind reviewer and another participant (other than editor) are selected.
+		// Editors are ignored, they can see everything.
+		$query = $this->getQuery();
+		// Queryies only support ASSOC_TYPE_SUBMISSION so far (see above)
+		if ($query->getAssocType() == ASSOC_TYPE_SUBMISSION) {
+			$request = Application::get()->getRequest();
+			$submissionId = $query->getAssocId();
+
+			$reviewAssignmentDao = DAORegistry::getDAO('ReviewAssignmentDAO'); /* @var $reviewAssignmentDao ReviewAssignmentDAO */
+			$stageAssignmentDao = DAORegistry::getDAO('StageAssignmentDAO'); /* @var $stageAssignmentDao StageAssignmentDAO */
+			$userGroupDao = DAORegistry::getDAO('UserGroupDAO'); /* @var $userGroupDao UserGroupDAO */
+
+			// get the selected participants
+			$newParticipantIds = $this->getData('users');
```

Comment:
Maybe want to cast this to `array` just in case there are no users submitted and previous validation doesn't check that.